From 82b113dfd1ffdcc6771d21898289168014559aff Mon Sep 17 00:00:00 2001
From: Dave Airlie <airlied@redhat.com>
Date: Mon, 14 Mar 2011 14:17:55 +1000
Patch-mainline: 2.6.39
References: fate#310916
Git-commit: bcd5023c961a44c7149936553b6929b2b233dd27
Subject: [PATCH 2261/2588] drm/i915: disable opregion lid detection for now.

At least on my HP 2540p this is wrong at bootup, fine
at any other time once a lid event has occured. This is due to
_REG vs _INI ordering in the ACPI tables.

Signed-off-by: Dave Airlie <airlied@redhat.com>
(cherry picked from commit bcd5023c961a44c7149936553b6929b2b233dd27)

Signed-off-by: Takashi Iwai <tiwai@suse.de>
---
 drivers/gpu/drm/i915/intel_panel.c |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_panel.c b/drivers/gpu/drm/i915/intel_panel.c
index 51a5354..a06ff07 100644
--- a/drivers/gpu/drm/i915/intel_panel.c
+++ b/drivers/gpu/drm/i915/intel_panel.c
@@ -284,18 +284,24 @@ void intel_panel_setup_backlight(struct drm_device *dev)
 enum drm_connector_status
 intel_panel_detect(struct drm_device *dev)
 {
+#if 0
 	struct drm_i915_private *dev_priv = dev->dev_private;
+#endif
 
 	if (i915_panel_ignore_lid)
 		return i915_panel_ignore_lid > 0 ?
 			connector_status_connected :
 			connector_status_disconnected;
 
+	/* opregion lid state on HP 2540p is wrong at boot up,
+	 * appears to be either the BIOS or Linux ACPI fault */
+#if 0
 	/* Assume that the BIOS does not lie through the OpRegion... */
 	if (dev_priv->opregion.lid_state)
 		return ioread32(dev_priv->opregion.lid_state) & 0x1 ?
 			connector_status_connected :
 			connector_status_disconnected;
+#endif
 
 	return connector_status_unknown;
 }
-- 
1.7.6

